packages/api/src/models/cleaningTaskModel.js

```diff
@@ -17,6 +17,67 @@ import Model from './baseFormatModel.js';
 import taskModel from './taskModel.js';
 
 class CleaningTaskModel extends Model {
+  // TODO: remove stub and update controller and FE to accept volume and weight
+  $parseDatabaseJson(json) {
+    // Remember to call the super class's implementation.
+    json = super.$parseDatabaseJson(json);
+    if (json.weight && json.weight_unit) {
+      json.product_quantity = json.weight;
+      json.product_quantity_unit = json.weight_unit;
+    } else if (json.volume && json.volume_unit) {
+      json.product_quantity = json.volume;
+      json.product_quantity_unit = json.volume_unit;
+    } else if (!json.volume && !json.weight) {
+      json.product_quantity = null;
+      json.product_quantity_unit = null;
+    }
+    // Database checks prevent quantity && !quantity_unit
+    delete json.weight;
+    delete json.weight_unit;
+    delete json.volume;
+    delete json.volume_unit;
+    return json;
+  }
+
+  // TODO: remove stub and update controller and FE to accept volume and weight
+  $formatDatabaseJson(json) {
+    // Remember to call the super class's implementation.
+    json = super.$formatDatabaseJson(json);
+    const weightUnits = ['g', 'lb', 'kg', 't', 'mt', 'oz'];
+    const volumeUnits = ['l', 'gal', 'ml', 'fl-oz'];
+    const defaultAction = (json) => {
+      json.volume = json.product_quantity;
+      //Database previously defaulted to 'l'
+      json.volume_unit = 'l';
+      json.weight = null;
+      json.weight_unit = null;
+      delete json.product_quantity;
+      delete json.product_quantity_unit;
+    };
+    if (json.product_quantity && json.product_quantity_unit) {
+      if (weightUnits.includes(json.product_quantity_unit)) {
+        json.weight = json.product_quantity;
+        json.weight_unit = json.product_quantity_unit;
+        json.volume = null;
+        json.volume_unit = null;
+        delete json.product_quantity;
+        delete json.product_quantity_unit;
+      } else if (volumeUnits.includes(json.product_quantity_unit)) {
+        json.volume = json.product_quantity;
+        json.volume_unit = json.product_quantity_unit;
+        json.weight = null;
+        json.weight_unit = null;
+        delete json.product_quantity;
+        delete json.product_quantity_unit;
+      } else {
+        defaultAction(json);
+      }
+    } else if (json.product_quantity && !json.product_quantity_unit) {
+      defaultAction(json);
+    }
```

Comment:
I think these cases can be combined: remove the check on `json.product_quantity_unit` in the first `if` and handle scenarios where `json.product_quantity_unit` either doesn't exist or isn't in `weightUnits` or `volumeUnits`.

Reply:
🙇  does it look better? Thanks!

Reply:
I love it, thank you!

Since `[1,2,3].includes(null)` returns `false`, the first check in the if below could be removed.
`if (json.product_quantity_unit && weightUnits.includes(json.product_quantity_unit))`

I'll approve, so please merge unless you want to remove it!

Reply:
Okay yeah I didn't look includes up but thought in case undefined. But that also results false too. 

I might just merge because this should all be deleted anyways!